### PR TITLE
[systemtest][cruise-control] Change testManuallyCreateMetricsReporterTopic

### PR DESCRIPTION
**Pull request by im-konge**
### Type of change

- Test update

### Description

After discussion with @ppatierno we decide to change our test `testManuallyCreateMetricsReporterTopic` to be more relevant. So now it is testing ability to recreate CC topics after deletion. The main problem here was that if you not set `auto.create.topics.enable` to `false` in Kafka, it will continuously create CC topics but with `partitions: 1, replication.factor: 1` -> the default setting for all Kafka topics. So I disabled it in both Kafka and CC, then remove this setting from Kafka and after that, I set the `auto.create.topics.enable` to `true` for CC. Interesting thing is that CC will create the `strimzi.cruisecontrol.modeltrainingsamples` and `strimzi.cruisecontrol.partitionmetricsamples` with settings: `partitions: 32, replication.factor: 2` only **if the topics are not created yet** -> the Kafka sometimes creates the topic even if you remove the `auto.create.topics.enable` from the config. If the topics are present, CC server will only change the `partitions: 32` and `replication.factor` will leave on `1`. That's why the asserts about `replication.factor` are commented.

Other than this feature I removed our method for creating CC with this config from `KafkaResource` -> I don't think it's really needed here, because it's used only in one test. Maybe we can add it in future if it will be worth.

I added the method for KafkaTopic deletion with wait to `KafkaTopicUtils`.

### Checklist

- [x] Write tests
- [x] Make sure all tests pass

### CODE REVIEW

**Comment by im-konge** [comment]
@strimzi-ci run tests test_only profile=regression testcase=CruiseControlIsolatedST#testAutoCreationOfCruiseControlTopics

File: systemtest/src/test/java/io/strimzi/systemtest/cruisecontrol/CruiseControlIsolatedST.java [hunk 1]
```diff
@@ -42,47 +42,65 @@
...
-            .editSpec()
-                .addToConfig("cleanup.policy", "delete")
+    void testAutoCreationOfCruiseControlTopics() throws InterruptedException {
+        KafkaResource.kafkaWithCruiseControl(CLUSTER_NAME, 3, 3)
+            .editOrNewSpec()
+                .editKafka()
+                    .addToConfig("auto.create.topics.enable", "false")
+                .endKafka()
+                .editCruiseControl()
+                    .addToConfig("auto.create.topics.enable", "false")
```
Comment (ppatierno) [inline systemtest/src/test/java/io/strimzi/systemtest/cruisecontrol/CruiseControlIsolatedST.java:53]: this flag doesn't make sense in CC configuration. It's a broker configuration.

**Comment by ppatierno** [comment]
it's not clear to me what should be tested.
If you want to test the auto-creation of the topics, it's done on the startup by the metrics reporter and the CC sampler component. So I think you should just test that they are created with right partitions and replication factor as we discussed offline.
Testing if they are re-created on deletion is a different test I think but as I mentioned the partitions and replication factors will be different. Anyway, I see a Kafka broker flag, used for configuring CC.
Can you elaborate a little bit more on what you are trying to achieve please?

**Comment by im-konge** [comment]
@ppatierno sorry, I read the documentation bad, I'm gonna remove it.

Anyway I think I can remake the test only if the partitions will be correctly set -> right partitions and right replication factor. It depends what do you prefer.

**Comment by ppatierno** [comment]
@im-konge for sure the test should test that on startup all the three topics are created correctly with the default partitions and replication factors you know (32, 1, ... bla bla bla) because they are created by the metrics reporter and CC on startup.
If you delete them, CC and metrics reporter won't be created them anymore but it will be the Kafka broker as soon as the metrics "stuff" will start to write/read from the not existing topics (you deleted) but then they will be created with the default Kafka broker partitions/replicas which are 1/1;all this stuff of course only if Kafka broker auto topic creation is enabled.
To be honest I wouldn't test this or you should expect that the partitions are not 32 :-)

**Comment by im-konge** [comment]
@ppatierno okay, I think it will be better just test the three topics on startup, no need to check it after the deletion, especially when the partitions/rep.factor will be 1/1 like a Kafka broker default.

Thanks a lot @ppatierno!

File: systemtest/src/main/java/io/strimzi/systemtest/utils/kafkaUtils/KafkaTopicUtils.java [hunk 2]
```diff
@@ -73,6 +73,12 @@ public static void waitForKafkaTopicCreationByNamePrefix(String topicNamePrefix)
         );
     }
 
+    public static void deleteKafkaTopicWithWait(String topicName) {
+        LOGGER.info("Deleting topic {} in namespace {}", topicName, kubeClient().getNamespace());
+        KafkaTopicResource.kafkaTopicClient().inNamespace(kubeClient().getNamespace()).withName(topicName).delete();
+        waitForKafkaTopicDeletion(topicName);
+    }
```
Comment (ppatierno) [inline systemtest/src/main/java/io/strimzi/systemtest/utils/kafkaUtils/KafkaTopicUtils.java:80]: do we still need this method?

Comment (im-konge) [reply systemtest/src/main/java/io/strimzi/systemtest/utils/kafkaUtils/KafkaTopicUtils.java:80]: Not at all, if we will need it in future, we add it in different PR, thanks :)

**Comment by im-konge** [comment]
@strimzi-ci run tests test_only profile=regression testcase=CruiseControlIsolatedST#testAutoCreationOfCruiseControlTopics

**Pull request merged**